File: ts/src/client_0430_runtime_ops.ts
import { compactRecord, type CCCC0430Client } from './client_0430_shared.js';
import type {
  WebModelRuntimeCompleteTurnOptions,
  WebModelRuntimeWaitNextTurnOptions,
} from './types.js';

export interface CCCC0430RuntimeOps {
  webModelRuntimeWaitNextTurn(
    options: WebModelRuntimeWaitNextTurnOptions
  ): Promise<Record<string, unknown>>;
  webModelRuntimeCompleteTurn(
    options: WebModelRuntimeCompleteTurnOptions
  ): Promise<Record<string, unknown>>;
}

const runtimeOps: CCCC0430RuntimeOps & ThisType<CCCC0430Client> = {
  async webModelRuntimeWaitNextTurn(options) {
    return this.call('web_model_runtime_wait_next_turn', {
      group_id: options.groupId,
      actor_id: options.actorId,
      by: options.by ?? options.actorId,
      limit: Math.min(Math.max(Math.trunc(options.limit ?? 20), 1), 20),
      kind_filter: options.kindFilter ?? 'all',
    });
  },

  async webModelRuntimeCompleteTurn(options) {
    return this.call('web_model_runtime_complete_turn', compactRecord({
      group_id: options.groupId,
      actor_id: options.actorId,
      by: options.by ?? options.actorId,
      turn_id: options.turnId,
      delivery_id: options.deliveryId,
      event_ids: options.eventIds,
      latest_event_id: options.latestEventId,
      status: options.status ?? 'done',
      summary: options.summary,
    }));
  },
};

export function installCCCC0430RuntimeOps(
  proto: CCCC0430Client & Partial<CCCC0430RuntimeOps>
): void {
  Object.assign(proto, runtimeOps);
}

File: python/src/cccc_sdk/client_0430_runtime_ops.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from .client_0430_shared import _compact


class CCCC0430RuntimeOpsMixin:
    """Web Model runtime operations supported by the current native contract."""

    def web_model_runtime_wait_next_turn(
        self,
        *,
        group_id: str,
        actor_id: str,
        by: Optional[str] = None,
        limit: int = 20,
        kind_filter: str = "all",
    ) -> Dict[str, Any]:
        aid = str(actor_id)
        return self.call(
            "web_model_runtime_wait_next_turn",
            {
                "group_id": str(group_id),
                "actor_id": aid,
                "by": str(by) if by is not None else aid,
                "limit": min(max(int(limit), 1), 20),
                "kind_filter": str(kind_filter),
            },
        )

    def web_model_runtime_complete_turn(
        self,
        *,
        group_id: str,
        actor_id: str,
        turn_id: str,
        delivery_id: str,
        event_ids: Optional[List[str]] = None,
        latest_event_id: str = "",
        status: str = "done",
        summary: str = "",
        by: Optional[str] = None,
    ) -> Dict[str, Any]:
        aid = str(actor_id)
        return self.call(
            "web_model_runtime_complete_turn",
            _compact(
                {
                    "group_id": str(group_id),
                    "actor_id": aid,
                    "by": str(by) if by is not None else aid,
                    "turn_id": str(turn_id),
                    "delivery_id": str(delivery_id),
                    "event_ids": [str(event_id) for event_id in event_ids]
                    if event_ids is not None
                    else None,
                    "latest_event_id": latest_event_id or None,
                    "status": str(status),
                    "summary": summary or None,
                }
            ),
        )
